.github/workflows/integration-test.yml

```diff
@@ -105,3 +104,41 @@ jobs:
         # For every other version and/or OS, run pytest without coverage
         if: (matrix.os != 'ubuntu-latest') || (matrix.python-version != env.TARGET_PYTHON_VERSION )
         run: poetry run pytest ${{ github.event.inputs.pytest_addopts }}
+  build:
+    name: Build versioned package
+    # This job is used to build the package and upload the artifacts
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup python, and load cache
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: ${{ env.TARGET_PYTHON_VERSION }}
+          cache-pre-commit: false
+          cache-venv: true
+          setup-poetry: true
+          install-deps: true
+      - name: Build
+        run: poetry build
+      - name: Get pycytominer version
+        id: get_version
+        run: |
+          echo "version=$(poetry version | cut -d' ' -f2 )"
+          echo "version=$(poetry version | cut -d' ' -f2 )" >> "$GITHUB_OUTPUT"
+      - name: Upload artifacts
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
```

Comment:
Consider using the semantic version specification for this action (instead of the hash-based reference) (here and below). If there's a reason for using the hash instead, consider adding a comment as to why.

Reply:
Good point, I can see how this might be helpful for security considerations! Candidly, I feel that GitHub Actions as an ecosystem are complex enough to warrant a "package manager" of some kind where the versions and reference locations are explicitly defined somewhere besides inline (especially when/if they are used many times). Would love to see if we could achieve something like this in the future for improved maintenance, security, and usability.

Reply:
Dependabot can also be configured to check your github action versions. That's probably the best cool that I'm aware of for doing that.